### PR TITLE
Remove audit components in favor of industry standard tools

### PR DESCRIPTION
**Pull request by scalvert**
As a further evolution of this addon, this PR removes the axe specific components that enable development integration in favor of other tools.

As per the README:

While this addon previously included a number of components that would aid in identifying axe violations during development, those have been deprecated in favor of other, industry standard tools such as:

- [**Lighthouse**](https://developers.google.com/web/tools/lighthouse) - an open-source, automated tool for improving the quality of web pages. You can run it against any web page, public or requiring authentication. It has audits for performance, accessibility, progressive web apps, SEO and more.
  - Homepage: https://developers.google.com/web/tools/lighthouse
- [**Sa11y**](https://ryersondmp.github.io/sa11y/) - an accessibility quality assurance tool that visually highlights common accessibility and usability issues. Geared towards content authors, Sa11y indicates errors or warnings at the source with a simple tooltip on how to fix.
- [axe Chrome extension](https://www.deque.com/axe/browser-extensions/) - a free axe browser extension ideal for development teams to test web applications to help identify and resolve common accessibility issues.

### CODE REVIEW

**Comment by MelSumner** [comment]
Lighthouse uses axe-core. Can we talk more about this please? I feel like this PR is changing how this addon works.

**Comment by scalvert** [comment]
@MelSumner sure. What specifically would you like to discuss? What are your concerns?

File: README.md [hunk 1]
```diff
@@ -196,171 +196,12 @@ export function a11yAuditIf(contextSelector, axeOptions) {
...
-ENV['ember-a11y-testing']: {
-  componentOptions: {
-    axeOptions: {
-      rules: {
-        'region': { enabled: true } // enable region rule
-      },
-      checks: {
-        'color-contrast': { options: { noScroll: true } } // disable scrolling of color-contrast check
-      }
-    }
```
Comment (drewlee) [inline README.md:353]: We should provide instructions on how to do this sort of thing in `test-helper.js` (assuming that's where it would go).

File: README.md [hunk 2]
```diff
@@ -196,171 +196,12 @@ export function a11yAuditIf(contextSelector, axeOptions) {
...
-#### Using Native aXe Global
-
-Ember A11y Testing injects the `axe` global during development and tests so that
-you can run accessibility audits while developing your application. It also
-provides some Ember-specific tools so that you can integrate accessibility
-checks into your workflow easily.
+- [**Lighthouse**](https://developers.google.com/web/tools/lighthouse) - an open-source, automated tool for improving the quality of web pages. You can run it against any web page, public or requiring authentication. It has audits for performance, accessibility, progressive web apps, SEO and more.
+  - Homepage: https://developers.google.com/web/tools/lighthouse
+- [**Sa11y**](https://ryersondmp.github.io/sa11y/) - an accessibility quality assurance tool that visually highlights common accessibility and usability issues. Geared towards content authors, Sa11y indicates errors or warnings at the source with a simple tooltip on how to fix.
+- [axe Chrome extension](https://www.deque.com/axe/browser-extensions/) - a free axe browser extension ideal for development teams to test web applications to help identify and resolve common accessibility issues.
```
Comment (drewlee) [inline README.md:204]:
```suggestion
- [**axe Chrome extension**](https://www.deque.com/axe/browser-extensions/) - a free axe browser extension ideal for development teams to test web applications to help identify and resolve common accessibility issues.
```

File: README.md [hunk 3]
```diff
@@ -196,171 +196,12 @@ export function a11yAuditIf(contextSelector, axeOptions) {
...
```
Comment (drewlee) [inline README.md:205]: We should also reference Microsoft's Accessibility Insights extension, which is the one we primarily use. I'd probably list it first.
```suggestion
[**Accessibility Insights for Web**](https://accessibilityinsights.io/docs/en/web/overview) - Accessibility Insights for Web helps developers find and fix accessibility issues in web apps and sites. This browser extension for Chrome and the new Microsoft Edge runs on Windows, MacOS, and Linux computers.

```

**Comment by rwjblue** [comment]
> Lighthouse uses axe-core.

Yes, indeed it does, that's why this is a good suggested replacement 😸.

> I feel like this PR is changing how this addon works.

This addon will become focused on testing only (as its name implies). It has been primarily focused on the testing aspects since its inception, and the introduction of the non-testing APIs adds non-trivial complexity that would be better served by a separate dedicated tool (either a separate addon, or usage of a tool that is not an Ember specific one).

File: README.md (hunk 3 above)
Comment (rwjblue) [reply README.md:205]: Ya, that's great to call out.

File: README.md (hunk 1 above)
Comment (rwjblue) [reply README.md:353]: Ya, it would basically be:

```js
// tests/test-helper.js

// ...snip...
import { setRunOptions } from 'ember-a11y-testing';

setRunOptions({
  rules: {
    region: { enabled: true },
  },
  checks: {
    'color-contrast': {
      options: {
        noScroll: true,
      },
    },
  },
});
```

File: README.md [hunk 4]
```diff
@@ -196,171 +196,12 @@ export function a11yAuditIf(contextSelector, axeOptions) {
 
 ### Development Usage
 
-Usage in development is restricted to applications using Ember 1.13 and up, as it
-relies on the `didRender` hook of a component's life-cycle (a feature only
-available in versions of Ember with the Glimmer rendering engine).
+While this addon previously included a number of components that would aid in identifying axe violations during development, those have been deprecated in favor of other, industry standard tools such as:
```
Comment (rwjblue) [inline README.md:199]: Let's reword this to be focused on the go forward users, not about migration to 4.0.0. We should author a custom transition guide for migrating from 3.x to 4.x and we can list the concrete things for those users in that guide.

**Comment by drewlee** [comment]
And probably nix the configuration here: https://github.com/ember-a11y/ember-a11y-testing/blob/c375436e166b61f0d2b158b71ec13ea37fab60db/tests/dummy/config/environment.js#L25

**Comment by drewlee** [comment]
We'll need to revamp the index page of the dummy app as it basically showcases the visual audit component feature.

File: README.md (hunk 3 above)
Comment (scalvert) [reply README.md:205]: Agree. Thanks for the callout, @drewlee!

File: README.md (hunk 1 above)
Comment (scalvert) [reply README.md:353]: Ah yes. Agree. This was weird as there was an odd interleaving of the documentation for the test-based code and the component-based code. I would have chosen to keep them very separate, but oh well.

I can update the README with this example that @rwjblue showed.

File: README.md (hunk 4 above)
Comment (scalvert) [reply README.md:199]: Sounds good.

**Comment by scalvert** [comment]
@drewlee I think as it pertains to the dummy apps' index route, we could just leave it as the acceptance tests use those elements. I think it's reasonable to remove any elements that aren't needed. WDYT?

**Comment by drewlee** [comment]
@scalvert Yeah, that's reasonable. I was thinking the revamp as something we can gradually work thru as opposed to something that needs to be done immediately. Thanks!

File: README.md [hunk 5]
```diff
@@ -194,6 +194,90 @@ export function a11yAuditIf(contextSelector, axeOptions) {
...
+```
+
+Test module level:
+
+```javascript
+import { module, test } from 'qunit';
+import { setRunOptions } from 'ember-a11y-testing';
+
+module('some test module', function (hooks) {
+  hooks.before(function () {
```
Comment (rwjblue) [inline README.md:224]: Needs to be `beforeEach` (this is because we reset after each test):

```suggestion
  hooks.beforeEach(function () {
```

File: README.md [hunk 6]
```diff
@@ -194,6 +194,90 @@ export function a11yAuditIf(contextSelector, axeOptions) {
...
+    // ...
+    a11yAudit();
+    // ...
+  });
+
+  // ...
+});
+```
+
+Options are automatically torn down when the test context's owner is destroyed.
```
Comment (rwjblue) [inline README.md:274]:
```suggestion
When using `setRunOptions` during a test, the options you set are automatically reset when the test completes.
```

File: addon-test-support/audit.ts [hunk 7]
```diff
@@ -111,10 +90,10 @@ export function _normalizeRunParams(
   }
 
   if (typeof options !== 'object') {
-    options = config[_configName]?.componentOptions?.axeOptions || {};
+    options = getRunOptions() || {};
```
Comment (rwjblue) [inline addon-test-support/audit.ts:93]: Why do we need to default? I think `axe-core` can handle `undefined` for options...

File: index.js [hunk 8]
```diff
@@ -36,7 +34,7 @@ module.exports = {
     );
 
     if (app.tests) {
-      let type = shouldExcludeAxeCore ? { type: 'test' } : undefined;
+      let type = options.excludeAxeCore ? { type: 'test' } : undefined;
```
Comment (rwjblue) [inline index.js:37]: I think this option is probably defunct now and should be removed?

File: README.md (hunk 5 above)
Comment (scalvert) [reply README.md:224]: Ya I originally was going to do this, but thought it made more sense to do it once rather than for each test invoked. I supposed it doesn't matter, but it's not clear to me what `getContext().owner` would return in `hooks.before`. What would it be?

File: addon-test-support/audit.ts (hunk 7 above)
Comment (scalvert) [reply addon-test-support/audit.ts:93]: This was the old behavior, so this was just preserving it.

File: index.js (hunk 8 above)
Comment (scalvert) [reply index.js:37]: Ya, I wasn't sure why it was there. I can remove.

File: addon-test-support/audit.ts (hunk 7 above)
Comment (scalvert) [reply addon-test-support/audit.ts:93]: I think I can probably remove it, since that param is optional anyway.

**Comment by scalvert** [comment]
Thanks, @drewlee! Really appreciate your careful review. I missed a bunch of things in the first iteration.
